src/middlewared/middlewared/plugins/account.py

```diff
@@ -2454,6 +2481,25 @@ async def __common_validation(self, verrors, data, schema, pk=None):
                     'You can\'t remove them.',
                 )
 
+        # Special handling for builtin_administrators group
+        if pk and 'users' in data:
+            group = await self.middleware.call('group.get_instance', pk)
+            if group['group'] == 'builtin_administrators':
+                # Get root user ID
+                root_user = await self.middleware.call(
+                    'user.query',
```

Comment:
This should be datastore.query since you're just getting the database primary key.